src/sidebar/components/test/user-menu-test.js

```diff
@@ -0,0 +1,268 @@
+        fakeServiceConfig.returns({ onProfileRequestProvided: true });
+
+        const wrapper = createUserMenu();
+
+        const profileMenuItem = findMenuItem(wrapper, fakeAuth.displayName);
+        assert.isFunction(profileMenuItem.prop('onClick'));
+      });
+
+      it('should be selectable if service does not support `onProfileRequest`', () => {
+        fakeServiceConfig.returns({ onProfileRequestProvided: false });
+
+        const wrapper = createUserMenu();
+
+        const profileMenuItem = findMenuItem(wrapper, fakeAuth.displayName);
+        assert.isFunction(profileMenuItem.prop('onClick'));
+      });
+    });
+
+    describe('profile-selected callback', () => {
+      it('should fire profile event when in sidebar for third-party user', () => {
+        UserMenu.$imports.$mock({
+          '../util/account-id': {
+            isThirdPartyUser: sinon.stub().returns(true),
+          },
+        });
+        const wrapper = createUserMenu();
+        const profileMenuItem = findMenuItem(wrapper, fakeAuth.displayName);
+        const onProfileSelected = profileMenuItem.prop('onClick');
+
+        onProfileSelected();
+
+        assert.equal(fakeBridge.call.callCount, 1);
+        assert.calledWith(fakeBridge.call, fakeProfileBridgeEvent);
+      });
+
+      it('should not fire profile event for first-party user', () => {
+        fakeIsThirdPartyUser.returns(false);
+        const wrapper = createUserMenu();
+        const profileMenuItem = findMenuItem(wrapper, fakeAuth.displayName);
+        const onProfileSelected = profileMenuItem.prop('onClick');
+
+        onProfileSelected();
+
+        assert.equal(fakeBridge.call.callCount, 0);
+      });
+
+      it('should not fire profile event when not in sidebar for third-party user', () => {
+        fakeInSidebar = false;
+        fakeIsThirdPartyUser.returns(true);
+        const wrapper = createUserMenu();
+        const profileMenuItem = findMenuItem(wrapper, fakeAuth.displayName);
+        const onProfileSelected = profileMenuItem.prop('onClick');
+
+        onProfileSelected();
+
+        assert.equal(fakeBridge.call.callCount, 0);
+      });
+    });
+  });
+
+  describe('account settings menu item', () => {
+    it('should be present if first-party user', () => {
+      fakeIsThirdPartyUser.returns(false);
+
+      const wrapper = createUserMenu();
+
+      const accountMenuItem = findMenuItem(wrapper, 'Account settings');
+      assert.isTrue(accountMenuItem.exists());
+      assert.calledWith(fakeServiceUrl, 'account.settings');
+    });
+
+    it('should not be present if third-party user', () => {
+      fakeIsThirdPartyUser.returns(true);
+
+      const wrapper = createUserMenu();
+
+      const accountMenuItem = findMenuItem(wrapper, 'Account settings');
+      assert.isFalse(accountMenuItem.exists());
+    });
+  });
+
+  describe('log out menu item', () => {
+    describe('logout menu item presence', () => {
+      const tests = [
```

Comment:
I pseudo-parametrized these tests but realized it isn't particularly readable so I didn't do it to the other tests in this module.